createAccountJava/src/main/java/com/brandon/createaccountjava/view/CreateAccountActivity.java

```diff
@@ -0,0 +1,192 @@
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.brandon.corejava.utilities.CustomDatePicker;
+import com.brandon.corejava.utilities.NotificationDialog;
+import com.brandon.createaccountjava.databinding.CreateAccountScreenBinding;
+import com.brandon.createaccountjava.viewmodel.CreateAccountUiStates;
+import com.brandon.createaccountjava.viewmodel.LegacyCreateAccountViewModel;
+import com.brandon.uicore.R;
+import com.google.android.material.textfield.TextInputEditText;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+@AndroidEntryPoint
+public class CreateAccountActivity extends AppCompatActivity {
+    @Inject
+    LegacyCreateAccountViewModel viewModel;
+    private Disposable disposable;
+    private CreateAccountScreenBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = CreateAccountScreenBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        final TextInputEditText firstNameField = binding.firstNameField;
+        final TextInputEditText lastNameField = binding.lastNameField;
+        final TextInputEditText emailField = binding.emailField;
+        final TextInputEditText dateField = binding.dateField;
+        final TextInputEditText locationField = binding.locationField;
+        final TextView firstError = binding.firstNameError;
+        final TextView lastError = binding.lastNameError;
+        final TextView emailError = binding.emailError;
+        final TextView dateError = binding.dateError;
+        final TextView locationError = binding.locationError;
+
+        onTextChanged(firstNameField, (String s) -> viewModel.firstNameChanged(s));
+        onTextChanged(lastNameField, (String s) -> viewModel.lastNameChanged(s));
+        onTextChanged(emailField, (String s) -> viewModel.emailChanged(s));
+        onTextChanged(dateField, (String s) -> viewModel.dateChanged(s));
+        onTextChanged(locationField, (String s) -> viewModel.locationChanged(s));
+
+        dateField.setOnClickListener(v -> new CustomDatePicker().openDialog(
+                this,
+                () -> viewModel.datePickerOpened(true),
+                (date) -> {
+                    dateField.setText((date.get(1) + 1) + "/" + date.get(2) + "/" + date.get(0));
+                    viewModel.datePickerOpened(false);
+                },
+                () -> viewModel.datePickerOpened(false)
+        ));
+
+
+        Objects.requireNonNull(binding.backArrow).setOnClickListener(v ->{} /* TODO FA-117 Navigate to Login */);
+
+        binding.createButton.setOnClickListener(v -> {
+            Objects.requireNonNull(binding.fieldsError).setVisibility(View.GONE);
```

Comment:
I don't see the reason to include `requireNonNull` anywhere in your PR. It will function the same as not doing anything

Reply:
adjusted to an inline check